android/src/main/java/flutter/plugins/contactsservice/contactsservice/ContactsServicePlugin.java

```diff
@@ -308,26 +319,62 @@ private void setAvatarDataForContactIfAvailable(Contact contact) {
     }
   }
 
-  private void loadContactPhotoHighRes(Contact contact, boolean photoHighResolution) {
-      try {
-          Uri uri = ContentUris.withAppendedId(ContactsContract.Contacts.CONTENT_URI, Long.parseLong(contact.identifier));
-          InputStream input = ContactsContract.Contacts.openContactPhotoInputStream(contentResolver, uri, photoHighResolution);
-
-          if (input == null){
-              contact.avatar = new byte[0];
-              return;
-          }
+  private void getAvatarForContacts(final List<String> identifiers, final boolean highRes,
+      final Result result) {
+    new GetAvatarsTask(identifiers, highRes, contentResolver, result).execute();
+  }
 
-          Bitmap bitmap = BitmapFactory.decodeStream(input);
-          input.close();
-          ByteArrayOutputStream stream = new ByteArrayOutputStream();
-          bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
-          contact.avatar = stream.toByteArray();
-          stream.close();
+  private static class GetAvatarsTask extends AsyncTask<Void, Void, List<byte[]>> {
+    final List<String> identifiers;
+    final boolean highRes;
+    final ContentResolver contentResolver;
+    final Result result;
+
+    GetAvatarsTask(final List<String> identifiers, final boolean highRes,
+        final ContentResolver contentResolver, final Result result) {
+      this.identifiers = identifiers;
+      this.highRes = highRes;
+      this.contentResolver = contentResolver;
+      this.result = result;
+    }
 
-      } catch (IOException e){
-          e.printStackTrace();
+    @Override
+    protected List<byte[]> doInBackground(final Void... params) {
+      // Load avatar for each contact identifier.
+      final List<byte[]> avatars = new ArrayList<>();
+      for (final String identifier : identifiers) {
+        avatars.add(loadContactPhotoHighRes(identifier, highRes, contentResolver));
       }
+
+      return avatars;
+    }
+
+    @Override
+    protected void onPostExecute(final List<byte[]> avatars) {
+      result.success(avatars);
```

Comment:
FYI: This can be called only once.

Reply:
I believe I only called `result.success()` once. Am I misusing this somehow?

Reply:
No action required here. It was an FYI (you are using it correctly).

Reply:
Gotcha, thanks for the heads up.